File: src/main/java/ru/ioleksiv/telegram/bot/core/model/method/group/GetChatAdministrators.java
package ru.ioleksiv.telegram.bot.core.model.method.group;

import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.method.interfaces.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.objects.std.ChatMember;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;

import java.util.List;

/**
 * @see <a href="https://core.telegram.org/bots/api#getchatadministrators">getChatAdministrators</a>
 */
public class GetChatAdministrators extends ChatAction<List<ChatMember>> {
    private static final String METHOD = "getChatAdministrators";

    protected GetChatAdministrators(Networker networker) {
        super(METHOD, networker);
    }

    @Override
    public Class<? extends CommonResponse<List<ChatMember>>> getResultWrapperClass() {
        return ResponseCollection.ChatMembersResponse.class;
    }

    @Override
    public GetChatAdministrators setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/group/GetChatAdministrators.kt
package io.github.oleksivio.tl.kbot.server.api.method.group

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.ChatMembersResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId
import io.github.oleksivio.tl.kbot.server.api.objects.std.ChatMember

/**
 * @see [getChatAdministrators](https://core.telegram.org/bots/api/#getchatadministrators)
 */
data class GetChatAdministrators(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    @JsonProperty(ApiDict.CHAT_ID_KEY)
    override val chatId: ChatId
) : ChatAction<List<ChatMember>>() {

    @JsonIgnore
    override val resultWrapperClass = ChatMembersResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getChatAdministrators"
}
